Tweaked insulated spray-on gloves

coiax opened the pull request:
:cl: coiax
tweak: Spray-on insulated gloves, a BEPIS tech, now produce a visible
message when disintegrating. That disintegration can be caused by
absorbing a shock, or now by washing them too much.
/:cl:

Some tweaking and code smoothing of the spray-on gloves. The only
mechanical change is in also letting them be washed off (although it
takes a while under a shower or using a sink).

A visible message also provides better feedback.

- Enforced `SHOULD_CALL_PARENT` of  `/atom/proc/wash()`.

coiax committed:
Tweaked insulated spray-on gloves

:cl: coiax
tweak: Spray-on insulated gloves, a BEPIS tech, now produce a visible
message when disintegrating. That disintegration can be caused by
absorbing a shock, or now by washing them too much.
/:cl:

Some tweaking and code smoothing of the spray-on gloves. The only
mechanical change is in also letting them be washed off (although it
takes a while under a shower or using a sink).

A visible message also provides better feedback.

coiax committed:
Make /wash a SHOULD_CALL_PARENT proc

A number of things depend on that signal being called.

Jared-Fogle commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -32,33 +32,31 @@
+	RegisterSignal(src, COMSIG_COMPONENT_CLEAN_ACT, .proc/use_charge)
+
+/obj/item/clothing/gloves/color/yellow/sprayon/proc/use_charge()
+	charges_remaining--
```
This needs SIGNAL_HANDLER.

Jared-Fogle commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -32,33 +32,31 @@
+/obj/item/clothing/gloves/color/yellow/sprayon/proc/use_charge()
+	charges_remaining--
+	if(charges_remaining <= 0)
+		var/turf/T = get_turf(src)
+		T.visible_message("<span class='warning'>[src] crumble[p_s()] away into nothing.</span>") // just like my dreams after working with .dm
```
```suggestion
		var/turf/location = get_turf(src)
		location.visible_message("<span class='warning'>[src] crumble[p_s()] away into nothing.</span>") // just like my dreams after working with .dm
```

Mothblocks reviewed (changes requested):
Fix that lint on lighting_object and whatnot as well, should be simple.

coiax committed:
Makes suggested changes

- Adds SIGNAL_HANDLER to signal handler
- Changes name of turf variable for readability
- Reenables lighting object override of parent
